src/main/java/io/github/moulberry/notenoughupdates/util/Utils.java

```diff
@@ -1798,4 +1811,50 @@ public static boolean isWithinRect(int x, int y, int left, int top, int width, i
 		return left <= x && x <= left + width &&
 			top <= y && y <= top + height;
 	}
+
+	public static int getNumberOfStars(ItemStack stack) {
+		if (stack != null && stack.hasTagCompound()) {
+			NBTTagCompound tag = stack.getTagCompound();
+
+			if (tag.hasKey("ExtraAttributes", 10)) {
+				NBTTagCompound ea = tag.getCompoundTag("ExtraAttributes");
+				if (ea.hasKey("upgrade_level", 99)) {
+					return ea.getInteger("upgrade_level");
+				} else if (ea.hasKey("dungeon_item_level")) {
+					return ea.getInteger("dungeon_item_level");
+				}
+			}
+		}
+		return 0;
+	}
+
+	public static String getStarsString(int stars, String displayName) {
+		//ignore master stars
+		if (displayName.contains("§c")) {
```

Comment:
while this behaviour might work in the current use case, since this is in `Utils` it would feel more appropiate to simply strip all stars (including master stars) from the display name and generate appropiate stars + master stars instead of returning null.

Reply:
This also breaks SUPREME items like Dungeon Heads: 
![image](https://user-images.githubusercontent.com/20768569/162978633-35d84bdc-c18f-435c-bae3-fbafe573e13e.png)
![image](https://user-images.githubusercontent.com/20768569/162978678-5b890257-f0fe-4d0c-b2a4-0c6bbc991d84.png)
![image](https://user-images.githubusercontent.com/20768569/162978723-412231f5-9c98-42ec-b02d-c4ca41bb2cbf.png)

Images taken on commit 0f4a6c5